tests/shakedown/shakedown/dcos/service.py

```diff
@@ -391,21 +394,36 @@ def service_unavailable_predicate(service_name):
     return response.status_code == 500
 
 
-def wait_for_service_endpoint(service_name, timeout_sec=120):
-    """Checks the service url if available it returns true, on expiration
-    it returns false"""
+def wait_for_service_endpoint(service_name, timeout_sec=120, path=""):
+    """
+    Checks the service url. Waits for exhibitor to start up (up to 20 minutes) and then checks the url on all masters.
+
+    if available it returns true,
+    on expiration throws an exception
+    """
+
+    def master_service_status_code(url):
+        logger.info('Querying %s', url)
+        auth = DCOSAcsAuth(dcos_acs_token())
+
+        response = requests.get(
+            url=url,
+            timeout=5,
+            auth=auth,
+            verify=verify_ssl())
+
+        return response.status_code
 
-    master_count = len(get_all_masters())
-    return time_wait(lambda: service_available_predicate(service_name),
-                     timeout_seconds=timeout_sec,
-                     required_consecutive_success_count=master_count)
+    schema = 'https' if ee_version() == 'strict' or ee_version() == 'permissive' else 'http'
+    logger.info('Waiting for service /service/{}/{} to become available on all masters'.format(service_name, path))
 
+    for ip in dcos_masters_public_ips():
+        url = "{}://{}/service/{}/{}".format(schema, ip, service_name, path)
+        assert_that(lambda: master_service_status_code(url), eventually(equal_to(200), max_attempts=timeout_sec/5))
```

Comment:
Why `timeout_sec/5`? `eventually` is defined as`(wait_fixed=1000, max_attempts=3)` - so here we'll wait only 24s?

Reply:
That's how it's defined on master https://github.com/mesosphere/marathon/blob/master/tests/system/common.py#L838.